app/src/main/java/com/example/mytestproject/ui/searchCity/SearchCityViewModel.kt

```diff
@@ -40,14 +39,15 @@ class SearchCityViewModel(
         _filteredCityList.value = cityFilter.filterCityList(inputResult)
     }
 
-    fun onCityChosen(cityId: Int) {
+    fun onCityChosen(cityId: Int, cityName: String) {
```

Comment:
You don't need to pass cityName from adapter into this method. Only id. You can get city name from filteredCityList using city id.

Reply:
Fixed